codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/SmithyNameResolver.java

```diff
@@ -62,13 +88,69 @@ public static String smithyTypesNamespaceAws(
     return serviceTrait.getSdkId().toLowerCase();
   }
 
+  public static String getSmithyType(
+    final Shape shape,
+    final Symbol symbol,
+    Model model,
+    SymbolProvider symbolProvider
+  ) {
+    if (
+      symbol.getNamespace().contains("smithy.") ||
+      symbol.getNamespace().equals("smithyapitypes") ||
+      knownSmithyType.contains(symbol.getName())
+    ) {
+      return symbol.getName();
+    }
+    if (shape.isMapShape()) {
+      MemberShape valueMemberShape = ((MapShape) shape).getValue();
+      Shape valueShape = model.expectShape(valueMemberShape.getTarget());
+      return "map[string]".concat(
+          SmithyNameResolver.getSmithyType(
+            valueShape,
+            symbolProvider.toSymbol(valueShape),
+            model,
+            symbolProvider
+          )
+        );
+    }
+    if (shape.isListShape()) {
+      MemberShape memberShape = ((ListShape) shape).getMember();
+      Shape memberShapeTarget = model.expectShape(memberShape.getTarget());
+      return "[]".concat(
+          SmithyNameResolver.getSmithyType(
+            memberShapeTarget,
+            symbolProvider.toSymbol(memberShapeTarget),
+            model,
+            symbolProvider
+          )
+        );
+    }
+    if (shape.isTimestampShape()) {
+      return "time"
```

Comment:
Add the Import for `time` with a TODO. This is not required for ESDK and `time` may not be the right type for `timestamp`

Reply:
shape visitor is a right place to add the import.